specs/eventing/test-plan/error-signalling-conformance.md

```diff
@@ -0,0 +1,470 @@
+     this from creating equality.Semantic differences (all other things held
+     constant).
+
+   message      <string>
+     A human readable message indicating details about the transition.
+
+   reason       <string>
+     The reason for the condition's last transition.
+
+   severity     <string>
+     Severity with which to treat failures of this type of condition. When this
+     is not specified, it defaults to Error.
+
+   status       <string> -required-
+     Status of the condition, one of True, False, Unknown.
+
+   type <string> -required-
+     Type of condition.
+```
+
+---
+
+## [Pre] Create a Broker instance which reconciliation fails
+
+```
+kubectl create -f specs/eventing/test-plan/control-plane/error-signalling/broker-not-ready.yaml
+```
+
+## [Test] Broker failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`False`, a severity of `""` (error), a non-empty `reason`, and a non-empty `message`:
+
+```
+kubectl get brokers/not-ready -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{";"}{.status.conditions[?(@.type=="Ready")].severity}{";"}{.status.conditions[?(@.type=="Ready")].reason}{";"}{.status.conditions[?(@.type=="Ready")].message}'
+```
+
+### [Output]
+
+```
+False;;ChannelTemplateFailed;Error on setting up the ChannelTemplate: Broker.Spec.Config configuration not supported, only [kind: ConfigMap, apiVersion: v1]
+```
+
+---
+
+## [Pre] Create a Trigger instance which reconciliation fails
+
+```
+kubectl create -f specs/eventing/test-plan/control-plane/error-signalling/trigger-not-ready.yaml
+```
+
+## [Test] Trigger failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`False`, a severity of `""` (error), a non-empty `reason`, and a non-empty `message`:
+
+```
+kubectl get triggers/not-ready -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{";"}{.status.conditions[?(@.type=="Ready")].severity}{";"}{.status.conditions[?(@.type=="Ready")].reason}{";"}{.status.conditions[?(@.type=="Ready")].message}'
+```
+
+### [Output]
+
+```
+False;;BrokerDoesNotExist;Broker "does-not-exist" does not exist
+```
+
+---
+
+## [Pre] Create a Channel instance which reconciliation fails
+
+```
+kubectl create -f specs/eventing/test-plan/control-plane/error-signalling/channel-not-ready.yaml
+```
+
+## [Test] Channel failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`Unknown` and a severity of `""` (error).
+
+```
+kubectl get channels/not-ready -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{";"}{.status.conditions[?(@.type=="Ready")].severity}'
+```
+
+### [Output]
+
+```
+Unknown;
+```
```

Comment:
If anyone has a suggestion for simulating a `Ready: False` with the InMemoryChannel instead of `Ready: Unknown`, please let me know.
It would help making this test slightly more relevant, since we could also assert that an actual error message and reason are propagated to the user.